### PR TITLE
feat(cat-voices): Add simple key derivation from seed phrase

### LINKED ISSUE #876: 🛠️ [TASK] : Create a simple function to generate a single 256 bit ed25519 private key from the key phrase, and its matching public key.
Part of #813

### PR DESCRIPTION
**Pull request by apskhem**
# Description

Added a simple key derivation from a 12-word seed phrase.

## Related Issue(s)

Closes #876 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/packages/catalyst_voices_models/lib/src/key_derivation.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import 'dart:typed_data';
+
+import 'package:catalyst_voices_models/src/seed_phrase.dart';
+import 'package:convert/convert.dart';
+import 'package:ed25519_hd_key/ed25519_hd_key.dart';
+
+class Ed25519KeyPair {
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/key_derivation.dart:7]: A class called Ed25519KeyPair should already be defined in catalyst_cardano_serialization package but that's not where we should add this code.

Temporarily I would put it inside SeedPhrase as a non-static method, the return type should be an instance of Ed25519KeyPair from catalyst_cardano_serialization package.

File: catalyst_voices/packages/catalyst_voices_models/test/seed_phrase_test.dart [hunk 2]
```diff
@@ -70,5 +70,25 @@ void main() {
       final expectedWords = mnemonic.split(' ');
       expect(seedPhrase.mnemonicWords, expectedWords);
     });
+
+    test('should generate key pair with different valid offsets', () async {
+      for (final offset in [0, 4, 28, 32, 64]) {
+        final keyPair = await SeedPhrase().generateKeyPair(offset);
+
+        expect(keyPair, isNotNull);
+        expect(keyPair.publicKey.bytes.length, equals(32));
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/test/seed_phrase_test.dart:79]: The length is already defined in Ed25519PublicKey.length

File: catalyst_voices/packages/catalyst_voices_models/test/seed_phrase_test.dart [hunk 3]
```diff
@@ -70,5 +70,25 @@ void main() {
...
+        expect(keyPair.privateKey.bytes.length, equals(32));
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/test/seed_phrase_test.dart:80]: The length is already defined in Ed25519PrivateKey.length

File: catalyst_voices/packages/catalyst_voices_models/lib/src/seed_phrase.dart [hunk 4]
```diff
@@ -65,6 +67,27 @@ class SeedPhrase {
   /// The mnemonic phrase as a list of individual words.
   List<String> get mnemonicWords => mnemonic.split(' ');
 
+  /// Generates an Ed25519 key pair from a seed.
+  ///
+  /// Throws an [RangeError] If the provided [offset] is negative or exceeds
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/seed_phrase.dart:72]:
```suggestion
  /// Throws a [RangeError] If the provided [offset] is negative or exceeds
```

File: catalyst_voices/packages/catalyst_voices_models/lib/src/seed_phrase.dart [hunk 5]
```diff
@@ -65,6 +67,27 @@ class SeedPhrase {
...
+  /// the length of the seed (64).
+  ///
+  /// [offset]: The offset is applied
+  /// to the seed to adjust where key derivation starts. It defaults to 0.
+  Future<Ed25519KeyPair> generateKeyPair([int offset = 0]) async {
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/seed_phrase.dart:77]: How about calling it `deriveKeyPair`?

Comment (apskhem) [reply catalyst_voices/packages/catalyst_voices_models/lib/src/seed_phrase.dart:77]: Yeah, make more sense.

**Review by dtscalac — APPROVED** [review]
LGTM

**Pull request merged**